app/src/main/java/com/hedvig/app/feature/insurance/ui/InsuranceAdapter.kt

```diff
@@ -214,23 +215,44 @@ class InsuranceAdapter(
                 retry: () -> Unit,
                 marketManager: MarketManager
             ) {
-                if (data !is InsuranceModel.TerminatedContractsHeader && data !is InsuranceModel.CrossSellHeader) {
+                if (data !is InsuranceModel.TerminatedContractsHeader) {
                     return invalid(data)
                 }
                 composeView.setContent {
                     HedvigTheme {
-                        Subheading(
-                            when (data) {
-                                InsuranceModel.CrossSellHeader ->
-                                    stringResource(R.string.insurance_tab_cross_sells_title)
-                                InsuranceModel.TerminatedContractsHeader ->
-                                    stringResource(R.string.insurances_tab_more_title)
-                                else -> ""
-                            }
+                        Subheading(stringResource(R.string.insurances_tab_more_title))
+                    }
+                }
+            }
+        }
+
+        class NotificationSubheadingViewHolder(composeView: ComposeView) : ViewHolder(composeView) {
+
+            private var data by mutableStateOf<InsuranceModel.CrossSellHeader?>(null)
```

Comment:
Is there any advantage to using this state-based approach over simply performing the `setContent` in the `bind`-method? 🤔

Reply:
Alright! I may have merged this PR in the meantime, so you'll have to create that in a separate PR 😉